packages/headless/src/controllers/triggers/headless-redirection-trigger.ts

```diff
@@ -0,0 +1,99 @@
+import {Schema, Value} from '@coveo/bueno';
+import {Engine} from '../../app/headless-engine';
+import {
+  ConfigurationSection,
+  RedirectionSection,
+} from '../../state/state-sections';
+import {configuration, redirection} from '../../app/reducers';
+import {buildController, Controller} from '../controller/headless-controller';
+import {loadReducerError} from '../../utils/errors';
+import {validateOptions} from '../../utils/validate-payload';
+import {logTriggerRedirect} from '../../features/redirection/redirection-analytics-actions';
+
+export interface RedirectionTriggerOptions {
+  /**
+   * The function used to handle whenever the `Redirection` controller's state's `redirectTo` value changes.
+   */
+  onRedirect(): void;
```

Comment:
The method could return the string of the `redirectTo`, but since it's already returned in the state, it's not really necessary, user will be able to access it.

Question of preference